### PR TITLE
Add Clojure

### LINKED ISSUE #1147: Request for support Clojure
Please support the syntax highlight for [Clojure](https://clojure.org/) language. Thank you.

### PR DESCRIPTION
**Pull request by tkjone**
Closes #1147

Implements the updates requested in #1214 and adds syntax highlighting for:

- thread-first (->) macro
- thread-last (->>) macro
- nil

Regarding the updates, you will notice that each of my commits addresses the comments made by @Golmote in #1214.  Further, each commit should be in the order of the requested fixes.  I apologize for not piggybacking directly off the other PR.

Let me know if there is anything else I can improve upon.

### CODE REVIEW

File: components/prism-clojure.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copied from https://github.com/jeluard/prism-clojure
+Prism.languages.clojure = {
+	comment: /;+.*/,
+	string: /"(?:\\.|[^\\"\r\n])*"/,
+	operator: /(::|[:|'])\b[a-zA-Z][a-zA-Z0-9*+!-_?]*\b/g, //used for symbols and keywords
```
Comment (Golmote) [inline components/prism-clojure.js]: Can you make the group non-capturing here? And also remove the `g` flag. You could add the `i` flag to shorten the regexp a bit, and use `\w` in the second character class. Finally, I think the last `-` is supposed to be escaped (or moved at the end).
The whole pattern could be something like:
`/(?:::|[:|'])\b[a-z][\w*+!?-]*\b/i`

**Comment by paladox** [comment]
cc @Golmote  wondering if you could review this please? :)

File: tests/languages/clojure/boolean_feature.test [hunk 2]
```diff
@@ -0,0 +1,13 @@
+false
+true
+
```
Comment (Golmote) [inline tests/languages/clojure/boolean_feature.test]: Could you add a test for `nil` here? Since it's handled in the `boolean` pattern.

File: tests/languages/clojure/keyword_feature.test [hunk 3]
```diff
@@ -0,0 +1,37 @@
+(def)
+(defmacro)
+(if)
+(do)
+(let)
+(var)
+(fn)
+
```
Comment (Golmote) [inline tests/languages/clojure/keyword_feature.test]: Ideally, we should add checks for each and every keyword here.

File: tests/languages/clojure/keyword_feature.test [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+----------------------------------------------------
+
+[
+	["punctuation", "("],
+  ["keyword", "def"],
```
Comment (Golmote) [inline tests/languages/clojure/keyword_feature.test]: Could you indent those lines with a tab, please?

File: components/prism-clojure.js (hunk 1 above)
Comment (tkjone) [reply components/prism-clojure.js]: Resolved

File: tests/languages/clojure/boolean_feature.test (hunk 2 above)
Comment (tkjone) [reply tests/languages/clojure/boolean_feature.test]: Resolved

File: tests/languages/clojure/keyword_feature.test (hunk 3 above)
Comment (tkjone) [reply tests/languages/clojure/keyword_feature.test]: I added about 70 new keywords here - a lot of the ones that were posing issues when I was testing e.g. `<=` and `>=`.  I am happy to fill out the rest, but I have a high degree of confidence that the ones I did not include will pass ;)

File: tests/languages/clojure/keyword_feature.test (hunk 4 above)
Comment (tkjone) [reply tests/languages/clojure/keyword_feature.test]: Finished 😌

**Comment by tkjone** [comment]
@Golmote Thanks for the review!  Let me know what you think 💪

**Comment by Golmote** [comment]
@tkjone Yay! LGTM!
